src/LibraryManager/Utilities/EncryptionUtility.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Microsoft.Web.LibraryManager.Utilities
+{
+    /// <summary>
+    /// Utility class for encrypting/decrypting data for the current user.
+    /// </summary>
+    internal static class EncryptionUtility
+    {
+        private static readonly byte[] EntropyBytes = Encoding.UTF8.GetBytes("LibraryManager");
+
+        public static string EncryptString(string value)
+        {
+            byte[] decryptedByteArray = Encoding.UTF8.GetBytes(value);
+            byte[] encryptedByteArray = ProtectedData.Protect(decryptedByteArray, EntropyBytes, DataProtectionScope.CurrentUser);
+            string encryptedString = Convert.ToBase64String(encryptedByteArray);
+            return encryptedString;
+        }
+
+        public static string DecryptString(string encryptedString)
+        {
+            byte[] encryptedByteArray = Convert.FromBase64String(encryptedString);
+            byte[] decryptedByteArray = ProtectedData.Unprotect(encryptedByteArray, EntropyBytes, DataProtectionScope.CurrentUser);
```

Comment:
Oof, good catch.  Per https://stackoverflow.com/questions/48208262/net-standard-2-0-and-system-security-cryptography-protecteddata-protect it sound like this will probably throw on another OS.  I'll see what I can do about this one...

Reply:
I tried it out.  When I try to use the --setEncrypted on a linux box, I get this error:

>jamlew@jamlew-ubuntu:~/test$ libman config --setEncrypted foo2=bar2
Windows Data Protection API (DPAPI) is not supported on this platform.

The good news: the error message isn't horrible.  At least it's a "not supported" type message.

The bad news: I don't know of a way to encrypt the password credential from .NET when running on Linux (or Mac).  I don't know how common it is to need this for proxy settings, having never been stuck behind one, so maybe we can just pass on this until we get more feedback?  I definitely do not feel comfortable falling back to storing a password in plaintext, or rolling our own password hashing.

Reply:
What we have is better than nothing and works at least on Windows so I agree that we should go with that for now. I believe most proxies don't actually need the password. We shall see. :)